lib/publiccloud/gcp_client.pm

```diff
@@ -36,8 +36,8 @@ has vault => undef;
 sub init {
     my ($self) = @_;
     # For now we support Vault and the credentials-microservice. Vault will be removed after a certain transition period
-    if (get_var('PUBLIC_CLOUD_CREDENTIALS_URL')) {
-        my $data = get_credentials(CREDENTIALS_FILE);
+    if (my $url = get_var('PUBLIC_CLOUD_CREDENTIALS_URL_GCP')) {
+        my $data = get_credentials($url, CREDENTIALS_FILE);
```

Comment:
```suggestion
        my $data = get_credentials($url . 'GCE.json', CREDENTIALS_FILE);
```

with keeping single variable

Reply:
same applies to other clients

Reply:
Yes, this is what I proposed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15044
And in the description of this PR you'll see why I don't like that :)
What about SAP or any other account using this code? They won't use `GCE.json` but something else..